GTG/core/timer.py

```diff
@@ -70,15 +72,59 @@ def time_changed(self):
         self.timeout_source = GObject.timeout_add_seconds(secs_to_refresh,
                                                           self.emit_refresh)
 
-    def set_configuration(self, refresh_hour, refresh_min):
+    def set_configuration(self, time):
+        self.config.set('hour', time.hour)
+        self.config.set('min', time.minute)
+
+    def get_configuration(self):
         try:
-            datetime.time(int(refresh_hour), int(refresh_min))
-        except (ValueError, TypeError):
-            raise ValueError("{}:{} is not a valid time".format(refresh_hour,
-                                                                refresh_min))
+            return datetime.time(int(self.config.get('hour')),
+                                 int(self.config.get('min')))
+        except(ValueError):
+            Log.error("Invalid time values. Try using HH:MM or am/pm format")
+            return datetime.time(0, 0)
 
-        self.config.set('hour', refresh_hour)
-        self.config.set('min', refresh_min)
+    def parse_time(self, time):
+        """
+        This function parses user input when setting
+        start of the day in general preferences.
+        input:  @time               user's input as string
 
-    def get_configuration(self):
-        return self.config.get('hour'), self.config.get('min')
+        output: @hour_value         default format for hours
+                @minute_value       default format for minutes
+                @ValueError         raised if @time is not valid
+        """
+        hour_value = ''
+        minute_value = ''
+        correct_hour, correct_min = False, False
+        # Format for HH:MM and optional am/pm
+        match = re.match(r'^(?P<hours>[0-9]{1,2}):?(?P<minutes>[0-9]{2})?( )?(?P<am_pm>[ap]m)?$', time)
+        if match:
+            hour_value = match.group('hours')
+            minute_value = match.group('minutes')
+            am_pm = match.group('am_pm')
+            if minute_value is None:
+                minute_value = '00'
+
+            correct_hour = 0 <= int(hour_value) < 24
+            correct_min = 0 <= int(minute_value) < 60
+
+            if am_pm == 'am':
+                correct_hour = 0 <= int(hour_value) < 13
+                if int(hour_value) == 12:
+                    hour_value = 0
+            elif am_pm == 'pm':
+                if 0 < int(hour_value) < 12:
+                    hour_value = str(int(hour_value) + 12)
+                    correct_hour = 11 < int(hour_value) < 24
+                elif int(hour_value) == 12:
+                    hour_value = 12
+                    correct_hour = True
+                elif int(hour_value) > 12:
+                    correct_hour = False
```

Comment:
This could be simplified:

```python
invalid_format = False
# ...
            elif am_pm == 'pm':
                if hour_value < 12:
                   hour_value += 12
                elif hour_value > 12:
                   invalid_format = True
```